src/tools/shorturl.py

```diff
@@ -0,0 +1,25 @@
+from abc import abstractmethod, ABCMeta
+from urllib import request
+
+
+class UrlShortener(metaclass=ABCMeta):
```

Comment:
What is your reasoning behind implementing the URL shortener as an object rather than the dynamic import module approach we used before?

Reply:
URL Shortening is generally very simple, I don't really think we need a whole module for every single provider. I found it more clear this way.